src/main.rs

```diff
@@ -1,3 +1,186 @@
+    let data: Value = serde_json::from_str(&methods).unwrap();
+    let obj = data.as_object().unwrap();
+    let cnt = obj.get("count").unwrap();
+    println!("{}", cnt);
+    let cnt = match cnt {
+        Value::Number(n) => n,
+        _other => panic!("other"),
+    };
+
+    let cnt = match cnt.as_u64() {
+        Some(x) => x,
+        None => panic!("None"),
+    };
+    if cnt == 0 {
+        return Vec::new();
+    }
+
+    let variants = obj.get("results").unwrap();
+
+    let variants = match variants.as_array() {
+        Some(v) => v,
+        None => panic!("None"),
+    };
+
+    let mut results: Vec<String> = Vec::new();
+    for variant in variants.iter() {
+        let s = variant.get("text_signature").unwrap().to_string();
+        println!("s.len: {}", s.len());
+        results.push(s[1..(s.len() - 1)].to_string());
+    }
+
+    results
+}
+
+struct Method {
+    full_name: String,
+    name: String,
+    args: Vec<String>,
+}
+
+fn process_method(method: String) -> Method {
+    let first_b = match method.find("(") {
+        Some(i) => i,
+        None => panic!("Invalid method"),
+    };
+    let last_b = match method.find(")") {
+        Some(i) => i,
+        None => panic!("Invalid method"),
+    };
+    let name = method[..first_b].to_string();
+    let args = method[first_b..last_b].split(',');
+    let args: Vec<&str> = args.collect();
+    let args: Vec<String> = args.iter().map(|&x| x.into()).collect();
+
+    Method {
+        name,
+        args,
+        full_name: method,
+    }
+}
+
+fn match_method_in_contract(code: &String, methods: Vec<String>) -> String {
+    let methods: Vec<Method> = methods
+        .iter()
+        .map(|m| process_method(m.to_string()))
+        .collect();
+    for method in methods {
+        let name_match = code.find(&method.name);
+        let name_match = match name_match {
+            Some(name) => name,
+            None => continue,
+        };
+        let first_b = name_match + method.name.len();
+        let last_b = match code[first_b..].find(")") {
+            Some(ind) => ind + first_b,
+            None => continue,
+        };
+        let mut flag = true;
+        for arg in method.args {
+            match code[first_b..last_b].find(&arg) {
+                Some(_) => continue,
+                None => {
+                    flag = false;
+                    break;
+                }
+            }
+        }
+        if flag {
+            return method.full_name;
+        }
+    }
+    "".to_string()
+}
+
+async fn index(contract_call: web::Json<ContractCall>) -> Result<String> {
+    let data = get_txn_info(&contract_call.txn).await;
+    let input_bytes = get_txn_input_bytes(&data);
+    let methods = get_txn_method(&input_bytes).await;
+    let methods = decode_methods(&methods);
+    let result = match_method_in_contract(&contract_call.contract, methods);
+    if result.len() == 0 {
+        return Ok("Method not found".to_string());
```

Comment:
also it is better to return not `Ok` but actix Error to make http status code 400 for example.